### ISSUE TITLE
Update cell classification network

### ISSUE THREAD

**Issue by adamltyson** [brainglobe/cellfinder#267]
For many reasons the existing cell classification network (and associated helper functions) is suboptimal:
* brainglobe/cellfinder-core#5 
* brainglobe/cellfinder-core#8
* brainglobe/cellfinder-core#13 

The use of TensorFlow is also causing headaches:
• https://github.com/brainglobe/cellfinder-core/issues/103
* https://github.com/brainglobe/cellfinder/issues/266

I think now is the time to fix this. Previously I've raised brainglobe/cellfinder-core#220 and brainglobe/cellfinder-core#221. But essentially my idea is to update the code (refactoring as we go) to Keras 3.0 [which is now out](https://github.com/keras-team/keras/releases/tag/v3.0.0). 

It would also be cool to address this in a way that would support:
* brainglobe/cellfinder-core#21

**Comment by sfmig** [comment]
First steps (chat with @IgorTatarnikov)
- migrate to Keras 3.0
   - verify accuracy of results (via tests)
   - verify no performance differences (via time/memory benchmarks)

**Comment by parikshit14** [comment]
Is there any reason for using Keras and PyTorch in the same package? It's generally not recommended because there is a lot of redundancy between them. Also, as they evolve, we will need to provide support for various versions.

**Comment by adamltyson** [comment]
Yes, Keras is the backend for PyTorch. I think it's fairly standard to use them together.

**Comment by adamltyson** [comment]
Worth pointing out to anyone interested in this, that we have already migrated to the latest versions of keras and pytorch, but the other open issues are still relevant
